OpenTrons/opentrons-api
# feat(protocol-designer,shared-data,components,app): adapter split support

closes RAUT-609 and RAUT-612
closes [RAUT-561](https://opentrons.atlassian.net/browse/RAUT-561) and [RQA-1076](https://opentrons.atlassian.net/browse/RQA-1076) and [RQA-1114](https://opentrons.atlassian.net/browse/RQA-1114)

# Overview

Adapter Split work in PD and App

Note: I will add more test coverage in a follow up pr

# Test Plan

For Pd:
sandbox: https://sandbox.designer.opentrons.com/pd_adapter-split/

1. import an older version PD protocol that has an adapter (one with H-s would work) and make sure that it migrates and exports successfully with the correct `loadLabware` commands
2. make a new protocol and add a H-S or Temperature module, make sure you can load the adapter + labware on top
3. mess around with the `moveLabware` form, move labware on/off the adapters and modules and make sure it properly errors if you try to move a labware onto an adapter/module that doesn't work with that labware. Also make sure the `movelabware` form allows you to select tipracks and not select the trash.

For the app:
1. upload a protocol that has an adapter, make sure the deck thumbnail loads the adapter + liquid information correctly. Check Protocol details in the labware and liquids tab. 
2. go to protocol setup and make sure labware setup, liquid setup list and map view show up correctly
3. go through LPC, should work as expected
4. run a protocol, look at the command text, it should display the adapter and labware on top of adapters text correctly
5. look through protocol and labware setup on the odd, map and list should display correctly

# Changelog

PD:
- add to `7_0_0` migration and the `fileCreator` to accommodate load adapter
- separate out the adapters in the deck setup to their own AdapterControls layer and change labware restrictions for H-S and Temperature module to allow for acceptable adapters
- add some more restrictions to moveLabware

Shared-data:
- added type `NonStackedLabware` and extended the `LabwareLocation` type to include `labwareId`.

App:
- added a bunch of logic to accommodate for if the labwareLocation has `labwareId` which included all the deck map renders in the app and odd (deckThumbnail, protocol setup, odd protocol setup) and the command text.

# Review requests

see test plan

# Risk assessment

low

[RAUT-561]: https://opentrons.atlassian.net/browse/RAUT-561?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RQA-1076]: https://opentrons.atlassian.net/browse/RQA-1076?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RQA-1114]: https://opentrons.atlassian.net/browse/RQA-1114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
